trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1924,4 +2204,190 @@ mod tests {
         let expected_index = bucket.get_index(&key).unwrap();
         assert_eq!(target_bucket_idx, expected_index as u8);
     }
+
+    #[rstest]
+    #[case(3, 3)]
+    #[case(7, 7)]
+    #[case(8, 8)]
+    #[case(17, 8)]
+    #[case(25, 8)]
+    fn test_limit_nodes_response_size(
+        #[case] original_nodes_size: usize,
+        #[case] correct_limited_size: usize,
+    ) {
+        let mut enrs: Vec<SszEnr> = Vec::new();
+        for _ in 0..original_nodes_size {
+            // Generates an ENR of size 63 bytes.
+            let (_, enr) = generate_random_remote_enr();
```

Comment:
Yea, this is brittle. What do you think about asserting that the size is what we think it is? That's probably shitty too, because it breaks in the exact same case. I'll think more about this.

Reply:
My first thought would be to build random ENRs that contain some meaningless key-value with [`add_value`](https://docs.rs/enr/0.6.0/enr/struct.EnrBuilder.html#method.add_value), so that the ENR size is random and less than or equal to the maximum size. With that list, you would pass it to the function and check that the size of the output list is within the proper bounds. You would also check that if there were any remaining (non-included) ENRs in the input, then adding the next ENR would indeed cause the size of the output list to breach the threshold.

I feel like the logic is straightforward though, so if the above turns out to not be so simple, then we can add more robust testing later.